frontend/src/app/plans/edit/[id]/_components/CreateNewPlanPage.tsx

```diff
@@ -80,11 +168,33 @@ export function CreateNewPlanPage({
     },
   });
 
+  const {
+    data: onlinePlan,
+    refetch: refetchOnlinePlan,
+    isLoading,
+  } = useQuery({
+    enabled: plan.onlineId !== null,
+    queryKey: ["onlinePlan", plan.onlineId],
+    queryFn: async () => {
+      const res = await getPlan({ id: Number(plan.onlineId) });
+      if (
+        res === false ||
+        (res as unknown as { status: number }).status === 404
+      ) {
+        plan.remove();
+        toast.error("Nie udało się pobrać planu");
+        router.push("/plans");
+        return null;
+      }
+      return res;
+    },
```

Comment:
error handling w queryFn? poczytaj sobie https://tkdodo.eu/blog/react-query-error-handling

Reply:
tu mi sie pomylilo, good

Reply:
poczytalem, jedyna przystepna opcja to dodanie nowego useeffecta czego juz nie chce robic